incubator/virtualcluster/pkg/syncer/patrol/differ/handler.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package differ
+
+import "sigs.k8s.io/multi-tenancy/incubator/virtualcluster/pkg/syncer/conversion"
+
+// HandlerFuncs is an adaptor to let you easily specify as many or
+// as few of the handle functions as you want while still implementing
+// Handler.
+type HandlerFuncs struct {
+	AddFunc    func(obj ClusterObject)
+	DeleteFunc func(obj ClusterObject)
+	UpdateFunc func(obj1, obj2 ClusterObject)
+}
+
+// OnAdd calls AddFunc if it's not nil.
+func (h HandlerFuncs) OnAdd(obj ClusterObject) {
+	if h.AddFunc != nil {
+		h.AddFunc(obj)
+	}
+}
+
+// OnUpdate calls UpdateFunc if it's not nil.
+func (h HandlerFuncs) OnUpdate(obj1, obj2 ClusterObject) {
+	if h.UpdateFunc != nil {
+		h.UpdateFunc(obj1, obj2)
+	}
+}
+
+// OnDelete calls DeleteFunc if it's not nil.
+func (h HandlerFuncs) OnDelete(obj ClusterObject) {
+	if h.DeleteFunc != nil {
+		h.DeleteFunc(obj)
+	}
+}
+
+// FilteringHandler applies the provided filter to all events coming
+// in. If any object match the filter, will skip this func.
+type FilteringHandler struct {
+	FilterFunc func(obj ClusterObject) bool
```

Comment:
It seems that we should have two FilterFuncs here: one for object 1 and one for object 2(in OnUpdate() call)

Reply:
i would like to keep it simpler. caller can distinguish them in on func.